sphinx/util/requests.py

```diff
@@ -50,6 +62,23 @@ def head(url: str, **kwargs: Any) -> requests.Response:
 
 
 class _Session(requests.Session):
+    def get_redirect_target(self, resp: requests.Response) -> str | None:
+        """Overrides the default requests.Session.get_redirect_target"""
+        # do not follow redirections that match ignored URI patterns
+        if resp.is_redirect:
+            location = resp.headers['location']
+            dest = urljoin(resp.url, location)
+            if any(pat.match(dest) for pat in self.to_ignore):
+                raise _IgnoredRedirection(
+                    location=location,
+                    status_code=resp.status_code,
+                )
+        return super().get_redirect_target(resp)
+
+    def __init__(self, *args: Any, **kwargs: Any) -> None:
+        self.to_ignore: list[re.Pattern[str]] = kwargs.pop('_to_ignore', [])
```

Comment:
Probably worth renaming this arg+attribute to something like `_ignored_redirects`; `_to_ignore` doesn't, in itself, describe much about what is intended for in the context of an HTTP session.